### PR TITLE
Terminology: Warn about use of 'master' in label expression

### PR DESCRIPTION
**Pull request by daniel-beck**
An attempt to make the label change more discoverable to regular users.

This warning currently only shows up when the chosen expression matches no nodes at all. More complex expressions may have unexpected behavior after the migration _without_ a warning appearing on the UI (because they will include nodes, probably even the built-in node in cases involving `!master`), but this should at least cover the most common case(s). We can always iterate later.

### Freestyle Job

> Screenshot 2021-08-19 at 20 05 25

### Pipeline Snippet Generator

> Screenshot 2021-08-19 at 20 05 40

> Screenshot 2021-08-19 at 20 12 24

> Screenshot 2021-08-19 at 20 13 28

### Proposed changelog entries

* Warn about use of `master` in a label expression when that's no longer in use

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by StefanSpieker — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

**Review by oleg-nenashev — APPROVED** [review]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Pull request merged**
